cli/src/opts/wallet/mod.rs

```diff
@@ -316,35 +322,18 @@ pub trait WalletTrait {
         Ok(builder.build()?)
     }
 
-    /// Attempts to find the actual path of the keystore file.
+    /// Ensures the path to the keystore exists.
     ///
-    /// If the path is a directory then we try to find the first keystore file with the correct
-    /// sender address
+    /// if the path is a directory, it bails and asks the user to specify the keystore file
+    /// directly.
     fn find_keystore_file(&self, path: impl AsRef<Path>) -> Result<PathBuf> {
         let path = path.as_ref();
         if !path.exists() {
             bail!("Keystore file `{path:?}` does not exist")
         }
 
         if path.is_dir() {
```

Comment:
Note that this is where compatibility with dapptools on this functionality is broken. Happy to accomodate suggestions on how we want to handle this as keystores do not have the expected `address` property.